src/iptc.cpp

```diff
@@ -345,22 +346,24 @@ namespace Exiv2 {
 
     void IptcData::printStructure(std::ostream& out, const Slice<byte*>& bytes, uint32_t depth)
     {
-        uint32_t i = 0;
-        while (i < bytes.size() - 3 && bytes.at(i) != 0x1c)
+        size_t i = 0;
+        while (i + 3 < bytes.size() && bytes.at(i) != 0x1c)
             i++;
         depth++;
         out << Internal::indent(depth) << "Record | DataSet | Name                     | Length | Data" << std::endl;
-        while (i < bytes.size() - 3) {
+        while (i + 3 < bytes.size()) {
```

Comment:
Hmm. I just realized that my logic that `i + 3` cannot overflow doesn't work here because `i` is incremented with a `+=` in this loop. I'll add another commit to fix it.